src/status_im/utils/keychain/core.cljs

```diff
@@ -60,29 +64,36 @@
 
 ;; Gets the password for a specified address from the Keychain
 (defn get-user-password [address callback]
-  (if-not platform/ios?
-    (callback) ;; no-op on Androids (for now)
-    (-> (.getInternetCredentials rn/keychain address)
-        (.then (partial handle-callback callback)))))
+  (-> (.getInternetCredentials rn/keychain address)
+      (.then (partial handle-callback callback))))
 
 ;; Clears the password for a specified address from the Keychain
 ;; (example of usage is logout or signing in w/o "save-password")
 (defn clear-user-password [address callback]
-  (if-not platform/ios?
-    (callback true)
-    (-> (.resetInternetCredentials rn/keychain address)
-        (.then callback))))
+  (-> (.resetInternetCredentials rn/keychain address)
+      (.then callback)))
 
 ;; Resolves to `false` if the device doesn't have neither a passcode nor a biometry auth.
 (defn can-save-user-password? [callback]
-  (if-not platform/ios?
-    (callback false)
+  (cond
+    platform/ios?
     (-> (.canImplyAuthentication
          rn/keychain
          (clj->js
           {:authenticationType
            (enum-val "ACCESS_CONTROL" "BIOMETRY_ANY_OR_DEVICE_PASSCODE")}))
-        (.then callback))))
+        (.then callback))
+
+    platform/android?
+    (-> (.getSecurityLevel rn/keychain)
+        (.then (fn [level]
+                 ;; we only enable password saving 
+                 ;; if secure hardware is available on Android
+                 (if (not= level keychain-secure-hardware)
+                   (callback false)
+                   (status/rooted-device? (comp callback not))))))
```

Comment:
Wouldn't this be more readable as:
```clojure
;; we only enable password saving 
;; if secure hardware is available on Android and the device is not rooted
(callback (and (= level keychain-secure-hardware)
               (not status/rooted-device?)))
```

Reply:
@PombeirP I think I made it more readable in a slightly different style ;)